### PR TITLE
fix(rpc): Filecoin.StateReplay

### PR DESCRIPTION
**Pull request by hanabi1224**
## Summary of changes

Changes introduced in this pull request:

- fix RPC method `Filecoin.StateReplay`
- API compare tests

```
     Running `target/quick/forest-tool api compare /home/me/fr/snapshots/calibnet/forest_snapshot_calibnet_2024-06-14_height_1700822.forest.car.zst --filter StateReplay`
| RPC Method                | Forest | Lotus |
|---------------------------|--------|-------|
| Filecoin.StateReplay (26) | Valid  | Valid |
```

## Reference issue to close (if applicable)

Closes

## Change checklist

- [x] I have performed a self-review of my own code,
- [x] I have made corresponding changes to the documentation,
- [x] I have added tests that prove my fix is effective or that my feature works (if possible),
- [x] I have made sure the [CHANGELOG][1] is up-to-date. All user-facing changes should be reflected in this document.

[1]: https://github.com/ChainSafe/forest/blob/main/CHANGELOG.md

### CODE REVIEW

File: src/message/mod.rs [hunk 1]
```diff
@@ -73,7 +73,7 @@ impl Message for ShimMessage {
         self.sequence = new_sequence;
     }
     fn required_funds(&self) -> TokenAmount {
-        &self.gas_fee_cap * self.gas_limit + &self.value
+        &self.gas_fee_cap * self.gas_limit
```
Comment (hanabi1224) [inline src/message/mod.rs:76]: [Lotus code](https://github.com/filecoin-project/lotus/blob/v1.27.0/chain/types/message.go#L126)
```go
func (m *Message) RequiredFunds() BigInt {
	return BigMul(m.GasFeeCap, NewInt(uint64(m.GasLimit)))
}
```

File: src/rpc/methods/state/types.rs [hunk 2]
```diff
@@ -79,6 +81,21 @@ pub struct MessageGasCost {
 
 lotus_json_with_self!(MessageGasCost);
 
+impl MessageGasCost {
+    pub fn try_new(message: &Message, apply_ret: &ApplyRet) -> anyhow::Result<Self> {
```
Comment (aatifsyed) [inline src/rpc/methods/state/types.rs:85]: nit: I think this can still be called `new`, even though it returns a `Result`, like `File::create`

**Review by aatifsyed — APPROVED** [review]

File: src/rpc/methods/state/types.rs (hunk 2 above)
Comment (hanabi1224) [reply src/rpc/methods/state/types.rs]: Fixed.

**Review by aatifsyed — APPROVED** [review]

File: src/state_manager/mod.rs [hunk 3]
```diff
@@ -582,7 +589,7 @@ where
             }
         };
         let result = self
-            .compute_tipset_state(Arc::clone(ts), Some(callback), VMTrace::NotTraced)
+            .compute_tipset_state(Arc::clone(ts), Some(callback), VMTrace::Traced)
```
Comment (ruseinov) [inline src/state_manager/mod.rs:592]: Do we want to keep `VMTrace::Traced` here?

**Comment by ruseinov** [comment]
Looks good, unsure about the trace.

File: src/state_manager/mod.rs (hunk 3 above)
Comment (hanabi1224) [reply src/state_manager/mod.rs:592]: `trace` is part of the response, see line 581
```rust
execution_trace:
```

**Review by lemmih — APPROVED** [review]

**Pull request merged**